File: dotnet/c#/SumEven/SumEven2/SumEven2.cs
using System;
using System.Linq;

class SumEven2
{
    static void Main()
    {
        int sumEven = Enumerable.Range(0, 101)      // 0 to 100 inclusive
            .Where(x => x % 2 == 0)                 // filter: only even numbers
            .Aggregate(0, (sum, x) => sum + x);     // reduce: sum the values

        Console.WriteLine(sumEven);
    }
}

File: java/SumEven2.java
import java.util.stream.IntStream;

public class SumEven2 {
    public static void main(String[] args) {
        int sumEven = IntStream.rangeClosed(0, 100)
                .filter(x -> x % 2 == 0)      // keep only even numbers
                .reduce(0, Integer::sum);     // sum them

        System.out.println(sumEven);
    }
}
